src/clients/inst_autoinit.rb

```diff
@@ -101,9 +101,16 @@ def main
         reg_enabled = reg_section["do_registration"]
 
         if !reg_enabled
-          msg = _("Registration is mandatory when using the online " \
-            "installation medium. Enable registration in " \
-            "the AutoYaST profile.")
+          if Yast::Mode.autoupgrade
+            msg = _("Registration is mandatory when using the online " \
+              "installation medium. Ensure that system is registered and " \
```

Comment:
Maybe I'd make it clear that the registration cannot be done during upgrade, something like: 
```suggestion
              "installation medium. Ensure that system is registered before running upgrade and " \
```